snuba/web/rpc/v1/resolvers/R_eap_items/storage_routing/routing_strategies/normal_mode_linear_bytes_scanned.py

```diff
@@ -0,0 +1,313 @@
+                    tier_specific_span.set_data(
+                        _SAMPLING_IN_STORAGE_PREFIX + "target_tier", target_tier
+                    )
+                    tier_specific_span.set_data(
+                        _SAMPLING_IN_STORAGE_PREFIX + "bytes_scanned_limit",
+                        bytes_scanned_limit,
+                    )
+            self.metrics.increment(
+                _SAMPLING_IN_STORAGE_PREFIX + "target_tier",
+                1,
+                {"tier": str(target_tier)},
+            )
+            span.set_data(_SAMPLING_IN_STORAGE_PREFIX + "target_tier", target_tier)
+            span.set_data(
+                _SAMPLING_IN_STORAGE_PREFIX + "estimated_target_tier_bytes_scanned",
+                estimated_target_tier_bytes_scanned,
+            )
+            routing_context.extra_info[
+                "estimated_target_tier_bytes_scanned"
+            ] = estimated_target_tier_bytes_scanned
+            return target_tier
+
+    def _run_query_on_most_downsampled_tier(
+        self, routing_context: RoutingContext
+    ) -> QueryResult:
+        with sentry_sdk.start_span(op="_run_query_on_most_downsampled_tier") as span:
+            routing_context.query_settings.set_sampling_tier(
+                self._get_most_downsampled_tier()
+            )
+            request_to_most_downsampled_tier = self._build_snuba_request(
+                routing_context
+            )
+            res = run_query(
+                dataset=PluggableDataset(name="eap", all_entities=[]),
+                request=request_to_most_downsampled_tier,
+                timer=routing_context.timer,
+            )
+
+            most_downsampled_query_bytes_scanned = self._get_query_bytes_scanned(res)
+            if most_downsampled_query_bytes_scanned == 0:
+                self.metrics.increment(
+                    "scanned_zero_bytes_in_most_downsampled_tier",
+                    1,
+                    {"tier": str(self._get_most_downsampled_tier())},
+                )
+
+            self._record_value_in_span_and_DD(
+                routing_context,
+                self.metrics.timing,
+                "query_bytes_scanned_from_most_downsampled_tier",
+                self._get_query_bytes_scanned(res, span),
+            )
+            return res
+
+    @with_span(op="function")
+    def _decide_tier_and_query_settings(
+        self, routing_context: RoutingContext
+    ) -> tuple[Tier, ClickhouseQuerySettings]:
+        if (
+            not routing_context.in_msg.meta.HasField("downsampled_storage_config")
+            or routing_context.in_msg.meta.downsampled_storage_config.mode
+            == DownsampledStorageConfig.MODE_UNSPECIFIED
+        ):
+            return Tier.TIER_1, {}
+
+        routing_context.query_result = self._run_query_on_most_downsampled_tier(
+            routing_context
+        )
+
+        query_settings = {
+            "max_execution_time": self._get_time_budget_ms() / 1000,
```

Comment:
this is wrong. We actually don't want to set this more strictly, we already set the max_execution time on the storage object, remove this setting from this strategy

Reply:
actually we can remove the settings entirely because we set timeout_overflow_mode=break on the storages as well